traffic_portal/app/src/common/directives/_directives.scss

```diff
@@ -0,0 +1,81 @@
+/*
+
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+*/
+
+.has-error {
+  div.search-box {
+    label {
+      color: #555555;
+      background-color: #eeeeee;
+      border-color: #ccc;
+    }
+    input {
+      box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
+      border-color: #ccc;
+    }
+  }
+}
+
+div.tree-select-root {
+  position: relative;
+
+  input.display-field {
+    background-color: #fff;
+  }
+
+  div.tree-drop-down {
+    width: 100%;
+    background: white;
+    position: absolute;
+    z-index: 2;
+    box-shadow: rgba(0, 0, 0, 0.38) 0px 3px 3px;
+    border: 1px solid #0000001f;
+    font-size: 16px;
+
+    div.search-box {
+      width: calc(100% - 10px);
+      margin: 5px auto 5px;
+
+      @media (min-width: 768px) {
+        div.helptooltip {
+          left: calc(100% - 50px);
+        }
+      }
+    }
+
+    ul.nav  {
+      max-height: 550px;
+      overflow-y: scroll;
+      overflow-x: hidden;
+
+      & > li.tree-row > a {
+        padding: 5px 0 5px 10px;
+
+
+        i {
+          margin-right: 3px;
+        }
+
+        @for $i from 0 through 30 {
+          &.depth-#{$i} {
+            position: relative;
+            left: #{$i * 8}px;
+          }
+        }
```

Comment:
This will break for sufficiently complex Tenancy trees - if you used a nested tree option component like

<details>
<summary><code>tree.select.tpl.html</code></summary>

```html
<div class="tree-select-root">
    <input type="text" id="{{ handle }}" name="{{ handle }}" class="form-control display-field" ng-model="selected.label" ng-click="toggle()" required readonly>
    <div class="tree-drop-down" ng-show="shown">
        <div class="input-group search-box">
            <label class="input-group-addon has-tooltip control-label" for="searchText">
               Filter:
               <div class="helptooltip">
                    <div class="helptext">Fuzzy search by tenant name</div>
                </div>
            </label>
            <input name="searchText" type="text" class="form-control" ng-model="searchText" maxlength="48">
        </div>
        <ul class="nav nav-list nav-pills nav-stacked">
            <treerow ng-repeat="row in treeRows | filter: checkFilters" row="row">
            </treerow>
        </ul>
    </div>
</div>
```

</details>

<details>
<summary><code>tree.row.tpl.html</code></summary>

```html
<li>
      <a ng-class="'depth-' + {{ row.depth }}" ng-click="select(row)" name="{{row.label}}">
          <i class="fa" ng-class="getClass(row)" ng-click="collapse(row, $event)"></i>
          <span>{{row.label}}</span>
       </a>
       <ul ng-if="children.length > 0">
           <tree-row row="child" ng-repeat="child in children"></treerow>
       </ul>
<li>
```

</details>

You could support arbitrary depths and also simply the styling a tiny bit with stacking margins

Reply:
I'd really rather avoid having to make another directive/component (which is the only way to do that in ajs). I'll move the `left` portion of the css to angular in an `ng-style`.